hetio/neo4j.py

```diff
@@ -281,6 +281,171 @@ def construct_dwpc_query(metarels, property='name', join_hint='midpoint', index_
 
     return query
 
+def construct_pdp_query(metarels, dwpc=None, property='name', join_hint='midpoint', index_hint=False, unique_nodes=True):
+    """
+    Create a cypher query for computing the path degree product for a type of path.
+    This function is very similar to construct_dwpc_query, with the main changes occuring in the
+    query's aggregation level.
+
+    Parameters
+    ----------
+    metarels : a metarels or MetaPath object
+        the metapath (path type) to create a query for
+    dwpc : int
+        the degree-weighted path count for the metapath. If dwpc is not provided,
+        a subquery will be added to calculate it.
+    property : str
+        which property to use for soure and target node lookup
+    join_hint : 'midpoint', bool, or int
+        whether to add a join hint to tell neo4j to traverse form both ends of
+        the path and join at a specific index. `'midpoint'` or `True` specifies
+        joining at the middle node in the path (rounded down if an even number
+        of nodes). `False` specifies not to add a join hint. An int specifies
+        the node to join on.
+    index_hint : bool
+        whether to add index hints which specifies the properties of the source
+        and target nodes to use for lookup. Enabling both `index_hint` and
+        `join_hint` can cause the query to fail.
```

Comment:
Is an integer `join_hint` equivalent to an `index_hint`?

Reply:
My understanding is that index hints tell the Cypher query planner to start searching at the supplied index, while join hints tell the planner to start in multiple places and meet at the join index. That being said, this was part of the code that I pulled from construct_dwpc_query, so you'd have to ask Daniel to be sure.